morepath/directive.py

```diff
@@ -1131,9 +1119,12 @@ def __init__(self, identity=object):
         '''
         self.identity = identity
 
-    def actions(self, obj):
-        yield FunctionAction(generic.verify_identity,
-                             identity=self.identity), obj
+    def identifier(self, reg_registry):
+        return self.identity
+
+    def perform(self, obj, reg_registry):
+        reg_registry.do_verify_identity.register(
```

Comment:
I now assume what you call ``reg_registry`` here is the scope registry itself. Since before the "Reg registry" was ``reg.Registry`` I think this naming is confusing and I'd call it the scope registry.